mempool/clist_mempool.go

```diff
@@ -646,28 +603,136 @@ func (mem *CListMempool) Update(
 	return nil
 }
 
+// recheckTxs sends all transactions in the mempool to the app for re-validation. When the function
+// returns, all recheck responses from the app have been processed.
 func (mem *CListMempool) recheckTxs() {
-	if mem.Size() == 0 {
-		panic("recheckTxs is called, but the mempool is empty")
+	if mem.Size() <= 0 {
+		return
 	}
+	mem.logger.Debug("recheck txs", "height", mem.height.Load(), "num-txs", mem.Size())
```

Comment:
I would move this debug log to the beginning of the function

Reply:
Done.